gensim/models/phrases.py

```diff
@@ -227,12 +243,17 @@ def export_phrases(self, sentences, out_delimiter=b' ', as_tuples=False):
             then you can debug the threshold with generated tsv
         """
         for sentence in sentences:
-            s = [utils.any2utf8(w) for w in sentence]
             last_bigram = False
             vocab = self.vocab
             threshold = self.threshold
             delimiter = self.delimiter  # delimiter used for lookup
             min_count = self.min_count
+
+            if self.recode_to_utf8:
+                s = [utils.any2utf8(w) for w in sentence]
+            else:
+                s = [utils.any2utf8(w) for w in sentence] if self.is_input_bytes else list(sentence)
```

Comment:
I think we shouldn't be recoding implicitly in `__getitem__`. The only gain is in the case when `learn_vocab` receives bytestrings, `recode_to_utf8` is `False` and `__getitem__` receives unicode. I don't think that justifies the dangerous subtle errors that it could cause.

Also, if we're going to be implicitly handling different input formats for `learn_vocab` vs `__getitem__`, we're also going to have to take care of the delimiter [here](https://github.com/RaRe-Technologies/gensim/pull/1413/files#diff-cb673cf37d57cdbaaa92b62f019a380dR263)